File: files/200-299/255/CH-255.R
library(tidyverse)
library(readxl)

path = "files/200-299/255/CH-255 Parse HTML.xlsx"
input = read_excel(path, range = "B2:B7")
test  = read_excel(path, range = "D2:G7")

parse_markup = function(text) {
  str_match_all(text, "<([^>]+)>([^<]+)</\\1>")[[1]] %>% 
    as_tibble(.name_repair = "unique") %>% 
    select(tag = 2, value = 3) %>%
    pivot_wider(names_from = tag, values_from = value) 
}

result = input %>%
  mutate(parsed = map(`Raw Text`, parse_markup)) %>%
  unnest(parsed) %>%
  select(-`Raw Text`) %>%
  set_names(names(test)) %>%
  mutate(across(c(ID, Value), as.numeric))

all.equal(result, test, check.attributes = FALSE) 
# TRUE

File: files/200-299/255/CH-255.py
import pandas as pd
import re

path = "200-299/255/CH-255 Parse HTML.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D:G", skiprows=1, nrows=6)

def parse_markup(text):
    matches = re.findall(r"<([^>]+)>([^<]+)</\1>", str(text))
    return {tag: value for tag, value in matches}

parsed = input["Raw Text"].apply(parse_markup).apply(pd.Series)
result = pd.concat([input, parsed], axis=1).drop(columns=["Raw Text"])
result.columns = test.columns
result[["ID", "Value"]] = result[["ID", "Value"]].apply(pd.to_numeric, errors="coerce")

print(result.equals(test))